src/V3MergeCond.cpp

```diff
@@ -0,0 +1,236 @@
+                            : new AstConst(rhsp->fileline(), AstConst::WidthedValue(), 1, 0);
+        } else if (AstNodeCond* const condp = extractCond(rhsp)) {
+            AstNode* const resp
+                = condTrue ? condp->expr1p()->unlinkFrBack() : condp->expr2p()->unlinkFrBack();
+            if (condp == rhsp) { return resp; }
+            if (AstAnd* const andp = VN_CAST(rhsp, And)) {
+                UASSERT_OBJ(andp->rhsp() == condp, rhsp, "Should not try to fold this");
+                return new AstAnd(andp->fileline(), andp->lhsp()->cloneTree(false), resp);
+            }
+        } else if (AstAnd* const andp = VN_CAST(rhsp, And)) {
+            if (andp->lhsp()->sameTree(m_mgCondp)) {
+                return condTrue ? andp->rhsp()->unlinkFrBack()
+                                : new AstConst(rhsp->fileline(), AstConst::WidthedValue(), 1, 0);
+            }
+            if (andp->rhsp()->sameTree(m_mgCondp)) {
+                return condTrue ? andp->lhsp()->unlinkFrBack()
+                                : new AstConst(rhsp->fileline(), AstConst::WidthedValue(), 1, 0);
+            }
+        }
+        rhsp->v3fatal("Don't know how to fold expression");
+    }
+
+    void mergeEnd() {
+        UASSERT(m_mgFirstp, "mergeEnd without list");
+        // Merge if list is longer than one node
+        if (m_mgFirstp != m_mgLastp) {
+            UINFO(6, "MergeCond - First: " << m_mgFirstp << " Last: " << m_mgLastp << endl);
+            ++m_statMerges;
+
+            // Create equivalent 'if' statement and insert it before the first node
+            AstIf* const ifp
+                = new AstIf(m_mgCondp->fileline(), m_mgCondp->unlinkFrBack(), NULL, NULL);
+            m_mgFirstp->replaceWith(ifp);
+            ifp->addNextHere(m_mgFirstp);
+            // Unzip the list and insert under branches
+            AstNode* nextp = m_mgFirstp;
+            do {
+                // Grab next pointer and unlink
+                AstNode* const currp = nextp;
+                nextp = currp != m_mgLastp ? currp->nextp() : NULL;
+                currp->unlinkFrBack();
+                // Skip over comments
+                if (VN_IS(currp, Comment)) {
+                    currp->deleteTree();
+                    continue;
+                }
+                // Count
+                ++m_statMergedItems;
+                // Unlink RHS and clone to get the 2 assignments (reusing currp)
+                AstNodeAssign* const thenp = VN_CAST(currp, NodeAssign);
+                AstNode* const rhsp = thenp->rhsp()->unlinkFrBack();
+                AstNodeAssign* const elsep = thenp->cloneTree(false);
+                // Construct the new RHSs and add to branches
+                thenp->rhsp(foldAndUnlink(rhsp, true));
+                elsep->rhsp(foldAndUnlink(rhsp, false));
+                ifp->addIfsp(thenp);
+                ifp->addElsesp(elsep);
+                // Cleanup
+                rhsp->deleteTree();
```

Comment:
```suggestion
                VL_DO_DANGLING(rhsp->deleteTree(), rhsp);
```

Reply:
I get that, but `rhsp` is `const`, so there is a conflict here between const annotation which I really like because it helps to read the code (well, helps me a lot anyways), and not having dangling poitners lying around. Same for other places.